client/Jammin/src/Pagination.jsx

```diff
@@ -0,0 +1,91 @@
+export default function Pagination({ currentPage, pageCount, onPageChanged, pageSize, onPageSizeChange, pageSizeOptions }) {
+  let pageNumberArray;
+
+  if (pageCount <= 6) {
+    pageNumberArray = [];
+    for (let i = 0; i < pageCount; i++) {
+      pageNumberArray.push(i + 1);
+    }
+  } else if (currentPage > 3 && currentPage < pageCount - 2) {
+    pageNumberArray = [
+      1,
+      null,
+      currentPage - 1,
+      currentPage,
+      currentPage + 1,
+      null,
+      pageCount,
+    ];
+  } else if (currentPage <= 3) {
+    pageNumberArray = [1, 2, 3, 4, null, pageCount];
+  } else {
+    pageNumberArray = [
+      1,
+      null,
+      pageCount - 3,
+      pageCount - 2,
+      pageCount - 1,
+      pageCount,
+    ];
+  }
+
+  const pageLinks = [];
+  pageNumberArray.forEach((pageNumber, index) => {
+    if (pageNumber === null) {
+      pageLinks.push(
+        <li key={index}>
+          <span className="pagination-ellipsis">&hellip;</span>
+        </li>
+      );
+    } else {
+      pageLinks.push(
+        <li key={index}>
+          <button
+            className={
+              "pagination-link " +
+              (pageNumber === currentPage ? "is-current" : "")
+            }
+            aria-label={`Go to page ${pageNumber}`}
+            onClick={() => onPageChanged(pageNumber)}
+          >
+            {pageNumber}
+          </button>
+        </li>
+      );
+    }
+  });
+
+  return (
+    <nav className="pagination" role="navigation" aria-label="pagination">
+      <button
+        className="pagination-previous is-family-code"
+        disabled={currentPage === 1}
+        onClick={() => onPageChanged(currentPage - 1)}
+      >
+        Previous
+      </button>
+      <button
+        className="pagination-next is-family-code"
+        disabled={currentPage === pageCount}
+        onClick={() => onPageChanged(currentPage + 1)}
+      >
+        Next page
+      </button>
+      <ul className="pagination-list is-family-code">{pageLinks}</ul>
+      <div className="select ml-1 mr-1">
+        <select
+          className="is-family-code"
+          id="tasks-per-page"
+          value={pageSize}
+          onChange={(e) => onPageSizeChange(Number(e.target.value))}
```

Comment:
It's good that you coerced the value into a number explicitly. 

You can leave it as is for now but perhaps an interesting read on why `parseInt` might be an even better option and why you'd want to add a check for `NaN` : https://journal.artful.dev/what-you-risk-when-using-number-to-parse-an-integer-from-a-string-in-typescript/